config/main.py

```diff
@@ -570,6 +570,102 @@ def is_ipaddress(val):
+    """ Check if port is already configured as mirror destination port """
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    for _,v in mirror_table.items():
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+def interface_has_mirror_config(mirror_table, interface_name):
+    """ Check if port is already configured with mirror config """
+    for _,v in mirror_table.items():
+        if 'src_port' in v and v['src_port'] == interface_name:
+            return True
+        if 'dst_port' in v and v['dst_port'] == interface_name:
+            return True
+
+    return False
+
+#
+# Check if SPAN mirror-session config is valid.
+#
+def validate_mirror_session_config(config_db, session_name, dst_port, src_port, direction):
+    """ Check if SPAN mirror-session config is valid """
+    if len(config_db.get_entry('MIRROR_SESSION', session_name)) != 0:
+        click.echo("Error: {} already exists".format(session_name))
+        return False
+
+    vlan_member_table = config_db.get_table('VLAN_MEMBER')
+    mirror_table = config_db.get_table('MIRROR_SESSION')
+    portchannel_member_table = config_db.get_table('PORTCHANNEL_MEMBER')
+    interface_table = config_db.get_table('INTERFACE')
+
+    if dst_port is not None:
+        if interface_name_is_valid(dst_port) is False:
+            click.echo("Error: Destination Interface {} is invalid".format(dst_port))
+            return False
+
+        if interface_is_in_vlan(vlan_member_table, dst_port):
+            click.echo("Error: Destination Interface {} has vlan config".format(dst_port))
+            return False
+
+        if interface_has_mirror_config(mirror_table, dst_port):
+            click.echo("Error: Destination Interface {} already has mirror config".format(dst_port))
+            return False
+
+        if interface_is_in_portchannel(portchannel_member_table, dst_port):
+            click.echo("Error: Destination Interface {} has portchannel config".format(dst_port))
+            return False
+
+        if interface_is_router_port(interface_table, dst_port):
+            click.echo("Error: Destination Interface {} is a L3 interface".format(dst_port))
+            return False
+
+    if src_port is not None:
+        for port in src_port.split(","):
+            if interface_name_is_valid(port) is False:
+                click.echo("Error: Source Interface {} is invalid".format(port))
+                return False
+            if dst_port is not None and dst_port == port:
+                click.echo("Error: Destination Interface cant be same as Source Interface")
+                return False
+            if interface_has_mirror_config(mirror_table, port):
+                click.echo("Error: Source Interface {} already has mirror config".format(port))
+                return False
+
+    if direction is not None:
+        if not any ( [direction == 'rx', direction == 'tx', direction == 'both'] ):
```

Comment:
Since we have a default, could this just be:
```suggestion
    if direction not in ['rx', 'tx', 'both']:
```

Reply:
Done. Thanks